sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/autorest.md

```diff
@@ -146,19 +146,19 @@ directive:
       }
     };
 
-# Move the stringIndexType parameter to the end for all operations referencing it.
+# Move the stringIndexType parameter to just before the newly append trainedModelLabel for all operations referencing it.
 - where-operation: ConversationalAnalysisAuthoring_Export
   transform: |
     var stringIndexTypeParamIndex = $.parameters.findIndex(param => param["$ref"] === "#/parameters/ConversationalAnalysisAuthoringStringIndexTypeQueryParameter");
     var stringIndexTypeParam = $.parameters[stringIndexTypeParamIndex];
+    $.parameters.splice(stringIndexTypeParamIndex, 1);
 
     var apiVersionParamIndex = $.parameters.findIndex(param => param["$ref"] === "common.json#/parameters/ApiVersionParameter");
     var apiVersionParam = $.parameters[apiVersionParamIndex];
+    $.parameters.splice(apiVersionParamIndex, 1);
 
-    $.parameters.splice(stringIndexTypeParamIndex, 1);
-    $.parameters.splice(apiVersionParamIndex - 1, 1);
-
-    $.parameters.push(stringIndexTypeParam);
+    var trainedModelLabelIndex = $.parameters.findIndex(param => param.name === "trainedModelLabel");
+    $.parameters.splice(trainedModelLabelIndex, 0, stringIndexTypeParam);
     $.parameters.push(apiVersionParam);
```

Comment:
@tg-msft could you take a look at this? I had to do something kind of gnarly to keep DPG parameter org consistent with our guidelines, especially since all the optional parameters are strings. The generated code, since the swagger parameter was inserted second to last (before api-version), effectively injected a parameter in the "middle" because you had me move `stringIndexType` last (well, before api-version, which we don't expose in the methods anyway).

I think this points out that we shouldn't mess with parameter order. If there's concern, we should make sure ordering issues are done in swagger or Cadl. I wouldn't expect autorest transforms like this (or whatever we'd use for Cadl) to be obvious, and straight-up auto-generating a client might not've caught this without test coverage (and maybe not even then depending on how the service responds).

/cc @m-nash

Reply:
One possibility at least with Cadl since we'd have some better versioning info is that if such a change is detected, perhaps we just generate a new method (keeping the old as-is) that uses an options class. Too bad there's no way to required named properties instead of having to support positional, too, but an options class would provide the same benefit.

Reply:
I talked a bit with @m-nash about this issue this morning, and he said this happens constantly in management and he's been thinking about mitigations. 👍🏻